From cbd8c7f8c5ff79f84ac952ca7a3bf6665be8fcc1 Mon Sep 17 00:00:00 2001
From: Chanh Nguyen <chanh@os.amperecomputing.com>
Date: Thu, 19 Dec 2024 09:14:19 +0000
Subject: [PATCH] Fix build error with boost 1.86

Added fix for boost::asio::spawn overload issue reported similar
to [1] during boost 1.86 migration build.

The proposed fix is to use default completion token.

[1] https://github.com/chriskohlhoff/asio/issues/1524

Upstream-Status: Pending
---
 src/system.hpp | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/system.hpp b/src/system.hpp
index d0af549..4a8071e 100644
--- a/src/system.hpp
+++ b/src/system.hpp
@@ -316,7 +316,8 @@ class DeviceMonitor
                     }
                 }
             }
-        });
+        },
+        {});
     }

     void addDevice(const NBDDevice& device)
@@ -430,7 +431,8 @@ class Process : public std::enable_shared_from_this<Process>
                    " Native: ", child.native_exit_code());

             onExit(child.exit_code());
-        });
+        },
+        {});
         return true;
     }

@@ -460,7 +462,8 @@ class Process : public std::enable_shared_from_this<Process>
                 child.terminate();
                 onTerminate();
             }
-        });
+        },
+        {});
     }

     std::string application()
--
2.34.1
